### PR TITLE
doc: VS Code dev setup

### PR DESCRIPTION
**Pull request by Kha**
* multi-root workspace
* default settings including .lean line length
* tasks `build` and `test`

### CODE REVIEW

File: lean.code-workspace [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			{
+				"label": "test",
+				"type": "shell",
+				"command": "NPROC=$(nproc 2>/dev/null || sysctl -n hw.logicalcpu 2>/dev/null || echo 4); CTEST_OUTPUT_ON_FAILURE=1 make -C build/release test -j$NPROC ARGS=\"-j$NPROC\"",
+				"problemMatcher": [],
+				"group": {
+					"kind": "test",
+					"isDefault": true
+				}
+			}
```
Comment (mhuisi) [inline lean.code-workspace:44]: These are very helpful. Would it be possible to add a Windows script as well using the `"windows": { "command": ... }` field, or do we let the Windows users take care of that?

File: lean.code-workspace [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		{
+			"path": "."
+		},
+		{
+			"path": "src"
+		}
+	],
+	"settings": {
+		"files.insertFinalNewline": true,
+		"files.trimTrailingWhitespace": true,
```
[the remark this answers is not in the document]
Comment (Kha) [reply lean.code-workspace:12]: This is straight from the old settings; fortunately we'll get a saner markup language in the future for user-facing docs so I think this is fine.

File: lean.code-workspace (hunk 1 above)
Comment (Kha) [reply lean.code-workspace:44]: I was kind of hoping that `nproc` would just work in msys but happy to leave that to others, yes.

File: doc/dev/index.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ del elan-init.ps1
 ```
 
 You can use `elan toolchain link` to give a specific stage build
-directory a reference name, then use `elan override set` to associate
+directory a reference name, then use a `lean-toolchain` file to associate
```
[the remark this answers is not in the document]
Comment (mhuisi) [reply doc/dev/index.md:34]: We decided to just add these files to the repo instead.

**Pull request merged**